seacrowd/sea_datasets/cc3m_35l/cc3m_35l.py

```diff
@@ -0,0 +1,244 @@
+
+    def is_target(self, line: dict, trg_lang: str):
+        if line["trg_lang"] == trg_lang:
+            return line
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        dev_path = dl_manager.download_and_extract(_URLS["trans_dev"])
+        train_path = dl_manager.download_and_extract(_URLS["trans_train"])
+
+        if self.config.data_dir is None:
+            raise ValueError("This is a local dataset. Please pass the data_dir kwarg to load_dataset.")
+        else:
+            data_dir = self.config.data_dir
+
+        # read tsv from local train and validation files
+        gcc_val = os.path.join(data_dir, "Validation_GCC-1.1.0-Validation.tsv")
+        gcc_train = os.path.join(data_dir, "Train_GCC-training.tsv")
+
+        # make it into pandas dataframe
+        colnames = ["caption", "img_url"]
+        gcc_val_df = pd.read_csv(gcc_val, sep="\t", header=None, names=colnames)
+        gcc_train_df = pd.read_csv(gcc_train, sep="\t", header=None, names=colnames)
+
+        # add new column to keep the downloaded image path
+        gcc_val_df["img_path"] = None
+        gcc_train_df["img_path"] = None
+
+        # add new column to keep the translated caption
+        gcc_val_df["trans_caption"] = None
+        gcc_train_df["trans_caption"] = None
+
+        gcc_val_df["backtrans_caption"] = None
+        gcc_train_df["backtrans_caption"] = None
+
+        # match the original captions in the translated set to the dataframe caption
+        # download the images from the URL and use it as the filepath
+        train_exceptions = []
+        val_exceptions = []
+
+        current_lang = self.config.subset_id.split("_")[2]
+        val_caption_targets = []
+        train_caption_targets = []
+
+        # filter validation data
+        with jl.open(os.path.join(dev_path), mode="r") as j:
+            val_caption_targets = [line for line in j if line["trg_lang"] == current_lang]
+            
+            #for line in val_caption_targets[:100]: # this was for debugging only
+            for line in val_caption_targets:
+                res = self.fill_img_path(gcc_train_df, line)
+                val_exceptions.extend(res[1])
+                gcc_val_df.update(res[0])
+            
+        # clean the memory
+        val_caption_targets = []
+
+        # filter train data
+        with jl.open(os.path.join(train_path), mode="r") as j:
+            train_caption_targets = [line for line in j if line["trg_lang"] == current_lang]
+            
+                
+            #for line in train_caption_targets[:100]: # this was for debugging only
+            for line in train_caption_targets:
+                res = self.fill_img_path(gcc_val_df, line)
+                train_exceptions.extend(res[1])
+                gcc_train_df.update(res[0])
+
+        # clean the memory
+        train_caption_targets = []
```

Comment:
Regarding this part, initially I had trouble with the extremely long loading time (more than days) as you reported. @SamuelCahyawijaya had tried to modify the code to support mapping (CMIIW) so the images are downloaded only when the data is fetched. The mapping (string matching part) was improved so it was not OOM. As a result the loading became much faster (downloading is a few hours at most, and loading takes minutes). Note that this dataloader should be run with `streaming=True` on the local test (not from the test module).

So, @IvanHalimP, would you mind if we update this part? In the meantime, could you clear the minor nits for a while?

Reply:
Sure thing!
Thank you @SamuelCahyawijaya for your help.